contributingGuides/TS_CHEATSHEET.md

```diff
@@ -0,0 +1,197 @@
+# Expensify TypeScript React Native CheatSheet
+
+## Table of Contents
+
+- [1.1 `props.children`](#children-prop)
+- [1.2 `forwardRef`](#forwardRef)
+- [1.3 Animated styles](#animated-style)
+- [1.4 Style Props](#style-props)
+- [1.5 Render Prop](#render-prop)
+- [1.6 Type Narrowing](#type-narrowing)
+- [1.7 Errors in Type Catch Clauses](#try-catch-clauses)
+
+## CheatSheet
```

Comment:
I think the example that @fabioh8010 prepared was really helpful, I would include it here or add it to `PROPTYPES_CONVERSION_TABLE.md` file (under the table):

```ts
// Before
const propTypes = {
    unknownData: PropTypes.any,
    anotherUnknownData: PropTypes.any,
    indexes: PropTypes.arrayOf(PropTypes.number),
    items: PropTypes.arrayOf(PropTypes.shape({
        value: PropTypes.string,
        label: PropTypes.string,
    })),
    shouldShowIcon: PropTypes.bool,
    onChangeText: PropTypes.func,
    count: PropTypes.number,
    session: PropTypes.shape({
        authToken: PropTypes.string,
        accountID: PropTypes.number,
    }),
    errors: PropTypes.objectOf(PropTypes.string),
    inputs: PropTypes.objectOf(PropTypes.shape({
        id: PropTypes.string,
        label: PropTypes.string,
    })),
    label: PropTypes.string,
    anchor: PropTypes.node,
    footer: PropTypes.element,
    uniqSymbol: PropTypes.symbol,
    icon: PropTypes.elementType,
    date: PropTypes.instanceOf(Date),
    size: PropTypes.oneOf(['small', 'medium', 'large']),
};

// After
type Item = {
    value: string;
    label: string;
};

type Session = {
    authToken: string;
    accountID: number;
};

type Input = {
    id: string;
    label: string;
};

type Size = 'small' | 'medium' | 'large';

type Props = {
    unknownData: string[];

    // It's not possible to infer the data as it can be anything because of reasons X, Y and Z.
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    anotherUnknownData: any;

    indexes: number[];
    items: Item[];
    shouldShowIcon: boolean;
    onChangeText: (value: string) => void;
    count: number;
    session: Session;
    errors: Record<string, string>;
    inputs: Record<string, Input>;
    label: string;
    anchor: React.ReactNode;
    footer: React.ReactElement;
    uniqSymbol: symbol;
    icon: React.ElementType;
    date: Date;
    size: Size;
};
```

Please note that you may encounter some PropTypes marked as `isRequired`, but when migrating to `type` you shall first consider all PropTypes as required ones and then inspect the component's source code and usage to identify which ones are really required or optional.

```ts
// Before
const propTypes = {
    isVisible: PropTypes.bool.isRequired,
    // `confirmText` prop is not marked as required here, theoretically it is optional.
    confirmText: PropTypes.string,
};

// After
type Props = {
    isVisible: boolean;
    // Consider it as required unless you have proof that it is indeed an optional prop.
    confirmText: string;
};
```

Reply:
added